Content.Server/_DV/CosmicCult/Abilities/CosmicFragmentationSystem.cs

```diff
@@ -0,0 +1,148 @@
+        {
+            _popup.PopupEntity(Loc.GetString("cosmicability-generic-fail"), ent, ent);
+            return;
+        }
+        if (HasComp<EmagSiliconLawComponent>(args.Target))
+        {
+            var doargs = new DoAfterArgs(EntityManager, ent, ent.Comp.CosmicSiphonDelay, new EventCosmicChantryDoAfter(), ent, args.Target)
+            {
+                DistanceThreshold = 2f,
+                Hidden = false,
+                BreakOnHandChange = true,
+                BreakOnDamage = true,
+                BreakOnMove = true,
+                BreakOnDropItem = true,
+            };
+            args.Handled = true;
+            _doAfter.TryStartDoAfter(doargs);
+        }
+        else if (HasComp<SiliconLawUpdaterComponent>(args.Target))
+        {
+            var doargs = new DoAfterArgs(EntityManager, ent, ent.Comp.CosmicSiphonDelay, new EventCosmicFragmentationDoAfter(), ent, args.Target)
+            {
+                DistanceThreshold = 2f,
+                Hidden = false,
+                BreakOnHandChange = true,
+                BreakOnDamage = true,
+                BreakOnMove = true,
+                BreakOnDropItem = true,
+            };
+            args.Handled = true;
+            _doAfter.TryStartDoAfter(doargs);
+        }
+
+        _cult.MalignEcho(ent);
+    }
+    private void OnCosmicChantryDoAfter(Entity<CosmicCultComponent> ent, ref EventCosmicChantryDoAfter args)
+    {
+        if (args.Args.Target is not { } target)
+            return;
+        if (args.Cancelled || args.Handled)
+            return;
+        args.Handled = true;
+        UnEmpower(ent);
+
+        var chantry = Spawn(ent.Comp.BorgChantry, Transform(target).Coordinates);
+        var polyVictim = _polymorph.PolymorphEntity(target, "CosmicFragmentationWisp");
+        if (polyVictim == null) // this really shouldn't ever be null but whatever
+            return;
+        EnsureComp<CosmicChantryComponent>(chantry, out var chantryComponent);
+        chantryComponent.PolyVictim = polyVictim.Value;
+        chantryComponent.Victim = target;
+    }
+    private void OnCosmicFragmentationDoAfter(Entity<CosmicCultComponent> ent, ref EventCosmicFragmentationDoAfter args)
+    {
+        if (args.Args.Target is not { } target)
+            return;
+        if (args.Cancelled || args.Handled)
+            return;
+        args.Handled = true;
+        UnEmpower(ent);
+
+        var lawboard = Spawn(ent.Comp.MalignLawboard, Transform(target).Coordinates);
+        _container.TryGetContainer(target, "circuit_holder", out var container);
+        if (container == null)
+            return;
+        _container.EmptyContainer(container, true);
+        _container.Insert(lawboard, container, Transform(target), true);
+    }
+
+    private void OnLawInserted(AILawUpdatedEvent args)
+    {
+        if (args.Lawset.Id == "CosmicCultLaws")
```

Comment:
If the only difference in these branches is the sendbriefing, then only put the sendbriefing behind the if